assets/javascripts/discourse/controllers/admin-plugins-filter-tag.js

```diff
@@ -6,24 +6,28 @@ import { popupAjaxError } from "discourse/lib/ajax-error";
 export default class extends Controller {
   @action
   setCategoryIdsForTag(filterTag, categories) {
-    let category_ids = [];
+    let categoryIds = [];
+    let categoriesAndParents = [];
 
+    categoriesAndParents.push(categories);
```

Comment:
If you change it into a Set from the start you won't need `new Set(categoriesAndParents.flat())` later (you'd have to change `push` calls to `add` too)

```suggestion
    let categoriesAndParents = new Set(categories);
```